troposphere/ec2.py

```diff
@@ -761,3 +762,58 @@ class VPCCidrBlock(AWSObject):
         'CidrBlock': (basestring, False),
         'VpcId': (basestring, True),
     }
+
+
+class TagSpecifications(AWSProperty):
+    props = {
+        'ResourceType': (basestring, False),
+        'Tags': ((Tags, list), False)
+    }
+
+
+class SpotOptions(AWSProperty):
+    props = {
+        'InstanceInterruptionBehavior': (basestring, False),
+        'MaxPrice': (basestring, False),
+        'SpotInstanceType': (basestring, False)
+    }
+
+
+class InstanceMarketOptions(AWSProperty):
+    props = {
+        'MarketType': (basestring, False),
+        'SpotOptions': (SpotOptions, False)
+    }
+
+
+class LaunchTemplateData(AWSProperty):
+    props = {
+        'BlockDeviceMappings': ([BlockDeviceMapping], False),
+        'CreditSpecification': (CreditSpecification, False),
+        'DisableApiTermination': (boolean, False),
+        'EbsOptimized': (boolean, False),
+        'ElasticGpuSpecifications': ([ElasticGpuSpecification], False),
+        'IamInstanceProfile': (IamInstanceProfile, False),
+        'ImageId': (basestring, True),
+        'InstanceInitiatedShutdownBehavior': (basestring, False),
+        'InstanceMarketOptions': (InstanceMarketOptions, False),
+        'InstanceType': (basestring, True),
```

Comment:
This isn't required either per the docs - really weird :)

Reply:
I verified through CFN execution that both InstanceType and ImageId are required

Reply:
Through more digging, AMI isn't required because when you make a new version of a LaunchTemplate it does not have to change.  I think that the way LaunchTemplates work with versioning is making the CFN really funky,  Its "versioning on top of versioning".

However when I execute the first time it says please create a proper LaunchTemplate with those two items specified, so at the very least its behavior is inconsistent.